File: AndroidAdvance/base-library/src/main/java/cn/xwj/base/BaseActivity.java
package cn.xwj.base;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import cn.xwj.ioc.inject.Injector;

/**
 * Author: xw
 * Date: 2018-03-01 14:03:45
 * Description: BaseActivity <this is description>.
 */

public abstract class BaseActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView();
        Injector.inject(this);
        initTitle();
        initView();
        initData();
    }

    public void actionStart(Class<? extends Activity> activity) {
        Intent intent = new Intent(this, activity);
        startActivity(intent);
    }

    protected abstract void initData();

    protected abstract void initView();

    protected abstract void initTitle();

    protected abstract void setContentView();


}

File: KotlinMail/baselibrary/src/main/java/cn/xwj/baselibrary/ui/activity/BaseActivity.kt
package cn.xwj.baselibrary.ui.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import cn.xwj.baselibrary.utils.AppManager

/**
 * Author: xw
 * Date: 2018-05-25 15:03:06
 * Description: BaseActivity: .
 */
open class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        AppManager.instances.addActivity(this)
    }

    override fun onDestroy() {
        super.onDestroy()
        AppManager.instances.removeActivity(this)
    }
}